### DISCUSSION TITLE
vscode custom devcontainer

### DISCUSSION THREAD

**Discussion by andyinno** [renovatebot/renovate#19425] [Request Help] [answered]
I am a bit lost, I googled a lot for understand how to solve this but I did not find anything.

basically I am using docker images that I get from mcr.microsoft.com in order to use them in a custom docker image that I build for all my projects development.

In detail I get mcr.microsoft.com/vscode/devcontainers/cpp:ubuntu-22.04@sha256:71ec005c8177823f4bd9fa299abc517d1d3b40228665f6293a01adc1d0cf78b3

and I have it in my dockerfile where I add other packages as a "common development image for cpp applications".

I would like to track changes to this image in order to be updated with the new published images.

What I tried until now are changes to my renovate.json in order to add a regexManager.
something like:

```
"regexManagers": [
        {
            "fileMatch": [
                "^Dockerfile$"
            ],
            "matchStrings": [
                "datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"
            ],
            "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
        },
        {
            "fileMatch": [
                "^Dockerfile$"
            ],
            "matchStrings": [
                "ARG IMAGE=(?<depName>.*?):[a-z-]*(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)\\s"
            ],
            "datasourceTemplate": "docker"
        }
    ]
```

This is the approach that I found in the regex manager examples.

What I got in the logs for this is 

```
Dependency mcr.microsoft.com/vscode/devcontainers/cpp has unsupported value ubuntu-22.04
```

Can I have some ideas about how to deal with it? also... I see that you are using a devcontainer folder in the renovate repo.
It does not use any mcr container. Should I switch to a default alpine/ubuntu/debian container on dockerhub for have it working?

thank you in advance

**Comment by rarkins** [comment, marked as answer]
I'm guessing that you are defaulting to `semver` versioning while the value is `ubuntu-22.04` which is not valid semver.

It's a long time until 24.04 so you don't really need that version updated soon, so that message is not much of a problem anyway. You should still get digest updates regardless

**Reply by andyinno** [reply]
Hello,

thank you for your reply.

I get your point. In two years or so I can update it manually or everything will be changed.

fair point.

I wil check for the digest update if triggers an update.

Just out of curiosity, what should the right way for dealing with version numbers similar to ubuntu-22.04? If the regexManager the right solution? I just need to tweak it a bit?

**Reply by rarkins** [reply]
In this case the best way would be to make sure that you extract just the `22.04` part as `currentValue` and then set versioning to be `ubuntu`. If you test that out in a dummy repo with e.g. `20.04` committed then you should then see a PR proposing `22.04`

**Reply by andyinno** [reply]
thanks!
